### PR TITLE
Bug 1583262 - Messaging router custom code to support about:welcome variants in 71

### PR DESCRIPTION
**Pull request by punamdahiya**

### CODE REVIEW

File: lib/ASRouter.jsm [hunk 1]
```diff
@@ -1009,38 +1020,43 @@ class _ASRouter {
...
-
-        // For the triplets experiment or non-experiment experience,
-        // we randomly assign a triplet and always use the "join" interrupt.
-      } else {
-        interrupt = "join";
-        triplet = await chooseBranch(
-          `${userId}-triplets-branch`,
-          TRAILHEAD_CONFIG.BRANCHES.triplets
-        );
+      if (
```
Comment (k88hudson) [inline lib/ASRouter.jsm]: Do we need to actually check for each of these variants to set triplet = supercharge? or can we set `triplet = "supercharge"` by default for all variants?

File: lib/ASRouter.jsm [hunk 2]
```diff
@@ -62,10 +62,21 @@ const TRAILHEAD_CONFIG = {
...
+      ["modal_variant_a"],
+      ["modal_variant_b"],
+      ["modal_variant_c"],
+      ["full_page_d"],
+      ["full_page_e"],
+    ],
     triplets: [["supercharge"], ["payoff"], ["multidevice"], ["privacy"]],
   },
-  LOCALES: ["en-US", "en-GB", "en-CA", "de", "de-DE", "fr", "fr-FR"],
   EXPERIMENT_RATIOS: [["", 0], ["interrupts", 1], ["triplets", 3]],
```
Comment (Mardak) [inline lib/ASRouter.jsm]: We probably don't want the "triplets" experiment using up 75% of the population. This should be 100% interrupts ?

File: lib/ASRouter.jsm (hunk 1 above)
Comment (punamdahiya) [reply lib/ASRouter.jsm]: Done

File: lib/ASRouter.jsm (hunk 2 above)
Comment (punamdahiya) [reply lib/ASRouter.jsm]: Done

File: lib/ASRouter.jsm [hunk 3]
```diff
@@ -62,11 +62,17 @@ const TRAILHEAD_CONFIG = {
...
+      ["modal_variant_b"],
+      ["modal_variant_c"],
+      ["full_page_d"],
+      ["full_page_e"],
+    ],
     triplets: [["supercharge"], ["payoff"], ["multidevice"], ["privacy"]],
   },
-  LOCALES: ["en-US", "en-GB", "en-CA", "de", "de-DE", "fr", "fr-FR"],
-  EXPERIMENT_RATIOS: [["", 0], ["interrupts", 1], ["triplets", 3]],
+  EXPERIMENT_RATIOS: [["", 0], ["interrupts", 1], ["triplets", 0]],
```
Comment (k88hudson) [inline lib/ASRouter.jsm]: Do you mind confirm with Felix that we do want 100% of users in an experiment branch (including 1 branch of control)?

**Comment by k88hudson** [comment]
This looks good to me assuming we want 100% of new users enrolled in one of these branches

File: lib/ASRouter.jsm (hunk 3 above)
Comment (punamdahiya) [reply lib/ASRouter.jsm]: NI Felix here https://bugzilla.mozilla.org/show_bug.cgi?id=1583262#c2

Comment (punamdahiya) [reply lib/ASRouter.jsm]: Scope of this bug is specific to about:welcome interrupts experiment in 71 and doesn't touch extended triplets experiment (hold back study) in 71. 

@k88hudson should we open a new bug to decide if we want to end extended triplets experiment in 71 which means we need to remove [setupExtendedTriplets here](https://github.com/mozilla/activity-stream/pull/5330/files#diff-6201bd23e10848d056ecc572cecd3e18R1057).  This should be follow up of https://bugzilla.mozilla.org/show_bug.cgi?id=1581912

Comment (punamdahiya) [reply lib/ASRouter.jsm]: Filed https://bugzilla.mozilla.org/show_bug.cgi?id=1589146

**Pull request merged**